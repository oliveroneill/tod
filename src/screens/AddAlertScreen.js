import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableHighlight,
  Button,
} from 'react-native';
import { List, ListItem } from 'react-native-elements'
import PropTypes from 'prop-types';

import TodAPI from '../utils/TodAPI.js';
import LoadingScreen from '../components/LoadingScreen.js';
import TouchableText from '../components/TouchableText.js';
import styles from '../style/Styles.js'

const list = [
  "Every Monday",
  "Every Tuesday",
  "Every Wednesday",
  "Every Thursday",
  "Every Friday",
  "Every Saturday",
  "Every Sunday"
]

class AddAlertScreen extends Component {
  static navigationOptions = ({ navigation }) => {
    const {state, setParams, navigate} = navigation;
    const params = state.params || {};
    const scheduleTrip = params.scheduleTrip || function(){};
    return {
      title: 'Schedule Trip',
      headerRight: (
        <TouchableText
          style={styles.navigationHeaderItem}
          onPress={scheduleTrip}
        >Set Alert</TouchableText>
      ),
    };
  };
  state = {
    loading: false,
    enabled: list.map(function(){return false;}),
  };
  constructor(props) {
    super(props);
    // bind functions
    this.onDayPressed = this.onDayPressed.bind(this);
    this.scheduleTrip = this.scheduleTrip.bind(this);
  }
  onDayPressed(i) {
    this.state.enabled[i]=!this.state.enabled[i];
    this.setState({enabled:this.state.enabled});
  }
  scheduleTrip() {
    let params = this.props.navigation.state.params;
    let {
      route,
      transport,
      inputArrival,
      inputArrivalDateString,
      origin,
      dest,
      waitingWindowMs,
      enabled
    } = params;
    this.setState({loading: true})
    this.api.scheduleTrip(
      origin, dest, route, transport, inputArrival,
      inputArrivalDateString, waitingWindow, enabled
    )
    .then(function() {
      alert("Trip successfully scheduled! Please note this is still a beta project");
      this.setState({loading:false});
      this.props.navigation.goBack();
    }.bind(this))
    .catch(function() {
      this.setState({loading:false});
      alert("Something went wrong. Please try again later");
    }.bind(this));
  }
  componentWillMount() {
    this.api = this.props.navigation.state.params.api;
    const { setParams } = this.props.navigation;
    setParams({scheduleTrip: this.scheduleTrip})
  }
  render() {
    let {route, transportIcon, subtitle} = this.props.navigation.state.params;
    return (
      <View style={styles.screen}>
        { this.state.loading ?
          <LoadingScreen
            loadingMessage="Scheduling Trip"
          />
        :
          <View style={styles.container}>
            <List containerStyle={{marginBottom: 20}}>
              <ListItem
                hideChevron={true}
                title={route.name}
                leftIcon={{name:transportIcon}}
                subtitle={subtitle}
                subtitleNumberOfLines={3}
              />
            </List>
            <Text style={{paddingLeft:10, fontSize:15, fontWeight:'bold'}}>Repeat:</Text>
            <List
              containerStyle={{marginBottom: 20}}>
              {
                list.map((l, i) => (
                  <ListItem
                    hideChevron={!this.state.enabled[i]}
                    rightIcon={{name:'done', style:{fontSize:19, color:'#027afe'}}}
                    key={i}
                    title={l}
                    onPress={() => this.onDayPressed(i)}
                    underlayColor='#e1e1e1'
                  />
                ))
              }
            </List>
          </View>
        }
      </View>
    )
  }
}

AddAlertScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}

export default AddAlertScreen;
