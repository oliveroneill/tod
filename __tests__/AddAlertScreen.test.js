import React from 'react';
import renderer from 'react-test-renderer';

import AddAlertScreen from '../src/screens/AddAlertScreen.js';

describe('AddAlertScreen', () => {
  let mockRowData = {
    route: {
      arrival_time: 1482363467071,
      departure_time: 1482363367071,
      description: "Take x avenue",
      name: "Bus Route 1"
    },
    subtitle: "Take x avenue",
    repeat_days: [false, true, true, false, false, false, true],
    transport_type:"walking",
    enabled: true,
    trip_id:"4543",
  };

  it('displays route correctly', () => {
    let component = renderer.create(
      <AddAlertScreen
        navigation={{state: {params: mockRowData}, setParams:function(){}}}
      />
    );
    expect(component.toJSON()).toMatchSnapshot()
  });
});