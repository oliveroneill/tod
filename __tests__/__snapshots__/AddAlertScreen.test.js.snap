// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAlertScreen displays route correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#bbb",
            "borderTopWidth": 1,
            "marginTop": 20,
          },
          Object {
            "marginBottom": 20,
          },
        ]
      }
    >
      <View
        disabled={false}
        onLongPress={undefined}
        onPress={undefined}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "#bbb",
              "borderBottomWidth": 1,
              "paddingBottom": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
            undefined,
            false,
            false,
          ]
        }
        underlayColor="white"
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#43484d",
                        "fontSize": 17.5,
                      },
                      false,
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Bus Route 1
              </Text>
            </View>
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={3}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#86939e",
                        "fontSize": 15,
                        "fontWeight": "600",
                        "marginTop": 1,
                      },
                      false,
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Take x avenue
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "bold",
          "paddingLeft": 10,
        }
      }
    >
      Repeat:
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#bbb",
            "borderTopWidth": 1,
            "marginTop": 20,
          },
          Object {
            "marginBottom": 20,
          },
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              undefined,
              false,
              false,
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#43484d",
                        "fontSize": 17.5,
                      },
                      Object {
                        "marginLeft": 10,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Every Monday
              </Text>
            </View>
            <View
              style={undefined}
            >
              <View />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              undefined,
              false,
              false,
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#43484d",
                        "fontSize": 17.5,
                      },
                      Object {
                        "marginLeft": 10,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Every Tuesday
              </Text>
            </View>
            <View
              style={undefined}
            >
              <View />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              undefined,
              false,
              false,
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#43484d",
                        "fontSize": 17.5,
                      },
                      Object {
                        "marginLeft": 10,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Every Wednesday
              </Text>
            </View>
            <View
              style={undefined}
            >
              <View />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              undefined,
              false,
              false,
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#43484d",
                        "fontSize": 17.5,
                      },
                      Object {
                        "marginLeft": 10,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Every Thursday
              </Text>
            </View>
            <View
              style={undefined}
            >
              <View />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              undefined,
              false,
              false,
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#43484d",
                        "fontSize": 17.5,
                      },
                      Object {
                        "marginLeft": 10,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Every Friday
              </Text>
            </View>
            <View
              style={undefined}
            >
              <View />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              undefined,
              false,
              false,
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#43484d",
                        "fontSize": 17.5,
                      },
                      Object {
                        "marginLeft": 10,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Every Saturday
              </Text>
            </View>
            <View
              style={undefined}
            >
              <View />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#bbb",
                "borderBottomWidth": 1,
                "paddingBottom": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
              undefined,
              false,
              false,
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Array [
                      Object {
                        "color": "#43484d",
                        "fontSize": 17.5,
                      },
                      Object {
                        "marginLeft": 10,
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Every Sunday
              </Text>
            </View>
            <View
              style={undefined}
            >
              <View />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
